sdk/python/feast/feature_store.py

```diff
@@ -1017,6 +1013,30 @@ def _augment_response_with_on_demand_transforms(
                     ] = GetOnlineFeaturesResponse.FieldStatus.PRESENT
         return OnlineResponse(GetOnlineFeaturesResponse(field_values=result_rows))
 
+    def _get_feature_views_to_use(
+        self,
+        features: Optional[Union[List[str], FeatureService]],
+        allow_cache=False,
+        hide_dummy_entity: bool = True,
+    ) -> List[FeatureView]:
+
+        passed_in_feature_views = (
+            {view.name: view for view in features.feature_views}
+            if isinstance(features, FeatureService)
+            else {}
+        )
```

Comment:
FYI lint complains here:
```
feast/feature_store.py:1126: error: "str" has no attribute "features"
feast/feature_store.py:1127: error: "str" has no attribute "name"
```
even though we establish using `isinstance` that `features` is a `FeatureService` and that `features.feature_views` can thus only be of type `List[FeatureView]`.... not sure what the best way to resolve this is.

Reply:
resolved, _group_feature_refs now only has to handle feature_ref strings as opposed to also FeatureServices